ts_src/block.ts

```diff
@@ -30,28 +30,141 @@ export class Block {
       return i;
     };
 
-    const readInt32 = (): number => {
-      const i = buffer.readInt32LE(offset);
-      offset += 4;
+    const readUInt8 = (): number => {
+      const i = buffer.readUInt8(offset);
+      offset += 1;
       return i;
     };
 
-    const block = new Block();
-    block.version = readInt32();
-    block.prevHash = readSlice(32);
-    block.merkleRoot = readSlice(32);
-    block.timestamp = readUInt32();
-    block.bits = readUInt32();
-    block.nonce = readUInt32();
-
-    if (buffer.length === 80) return block;
+    // const readInt32 = (): number => {
+    //   const i = buffer.readInt32LE(offset);
+    //   offset += 4;
+    //   return i;
+    // };
 
     const readVarInt = (): number => {
       const vi = varuint.decode(buffer, offset);
       offset += varuint.decode.bytes;
       return vi;
     };
 
+    const block = new Block();
+    block.version = readUInt32();
```

Comment:
I would use the `BufferReader` from `bufferutils.ts` to read the bytes instead of read* functions.